pkg/catalog/catalog.go

```diff
@@ -65,11 +60,49 @@ func GetDevfileRegistries(registryName string) (map[string]Registry, error) {
 	return devfileRegistries, nil
 }
 
+// convertURL converts GitHub regular URL to GitHub raw URL, do nothing if the URL is not GitHub URL
+// For example:
+// GitHub regular URL: https://github.com/elsony/devfile-registry/tree/johnmcollier-crw
+// GitHub raw URL: https://raw.githubusercontent.com/elsony/devfile-registry/johnmcollier-crw
+func convertURL(URL string) (string, error) {
+	url, err := url.Parse(URL)
+	if err != nil {
+		return "", err
+	}
+
+	if strings.Contains(url.Host, "github") && !strings.Contains(url.Host, "raw") {
+		// Convert path part of the URL
+		URLSlice := strings.Split(URL, "/")
+		if URLSlice[len(URLSlice)-2] == "tree" {
```

Comment:
Please check if the length is more then 2 so that the subtraction doesn’t yield a negative number

Reply:
I can add that validation but that case is unlikely because the format of regular GitHub URL is `<protocol>://<host>/<user>/<repository>`, it may or may not have branch part to append the URL, so definitely the length of the slice is more than 2.